### ISSUE TITLE
Import theme scss multiple times

### ISSUE THREAD

**Issue by jaetea** [angular/material2#4357]
#### Bug, feature request, or proposal:
Styling components

#### What is the expected behavior?
Per docs The theme file should not be imported into other SCSS files.

#### What is the current behavior?
Need to import theme file into component theme in order to style it correctly. Per docs
Now you just have to call the @mixin function to apply the theme:

// Import a pre-built theme
@import '~@angular/material/prebuilt-themes/deeppurple-amber.css';
// Import your custom input theme file so you can call the custom-input-theme function
@import 'app/candy-carousel/candy-carousel-theme.scss';

// Using the $theme variable from the pre-built theme you can call the theming function
@include candy-carousel-theme($theme);

#### What are the steps to reproduce?

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?

I have a custom global theme and need to implement this custom theme into a few custom components also.

#### Is there anything else we should know?

I'm struggling to understand how to have a global theme and import that into a custom components. The docs aren't helping as the file names in the examples don't match up and its not clear what the steps are.

**Comment by willshowell** [comment]
Can you be more specific as to which examples don't match up so they can be fixed?

I agree it's a little hard to follow, and you likely have to read over it twice to get it all. From **theming-you-own-components**, you can use a `@mixin` to theme your component. See the material components source for a supplementary guide of how this is done.

Alternatively, you can skip the `@mixin` approach and just consume the theme colors like
```
// my-custom.component.scss

@import '~@angular/material/theming';
@import 'my-theme-file';

$primary: map-get($theme, primary);
:host {
  background-color: mat-color($primary);
}
```
There are two caveats with this approach:
1. `my-theme-file` should *not* contain `mat-core()`. It should only ever be included once.
2. I don't think multiple/dynamic themes will work with this approach (please correct if this is wrong)

In order to keep `mat-core` from being included twice, you'll have to keep `my-theme-file` clear of anything but variable definitions. Call `mat-core` and `angular-material-theme` in another file that imports `my-theme-file`.

**Comment by jaetea** [comment]
When they specify in the second step
```
// Import your custom input theme file so you can call the custom-input-theme function
@import 'app/candy-carousel/candy-carousel-theme.scss';
```

Are they referring to this file in step one?

```
All you need is to create a @mixin function in the custom-component-theme.scss (candy-carousel-theme?)

// Import all the tools needed to customize the theme and extract parts of it
@import '~@angular/material/theming';

// Define a mixin that accepts a theme and outputs the color styles for the component.
@mixin candy-carousel-theme($theme) {
  // Extract whichever individual palettes you need from the theme.
  $primary: map-get($theme, primary);
  $accent: map-get($theme, accent);

  // Use mat-color to extract individual colors from a palette as necessary.
  .candy-carousel {
    background-color: mat-color($primary);
    border-color: mat-color($accent, A400);
  }
}
```

**Comment by willshowell** [comment]
1. Yes - the file in step one should be named something like `_candy-carousel-theme.scss` and only include that mixin that only affects the color-based styles of your component.

2. Call it from whatever file you call `angular-material-theme()` from:
```
...

// Include themes for all material components
@include angular-material-theme($my-custom-theme);

// Include themes for custom components
@include candy-carousel-theme($my-custom-theme);
```

**Comment by jelbourn** [comment]
The key takeaway is: don't import the file where you `@include` theming mixins. Anything else is fine to import. 

It's important to remember that the theming is really nothing but:
* Functions to define / consume palettes (which have no css output on their own)
* Mixins for each component that output the theme-specific styles
* Final `angular-material-theme` mixin that includes all of the individual component mixins. 

Doing your own theming just follows this pattern. 

[Here's a good tutorial for sass mixins.](https://scotch.io/tutorials/how-to-use-sass-mixins)

If you have any suggestions on how to improve the docs, feel free to open a PR.

**Comment by jaetea** [comment]
`// Define the palettes for your theme using the Material Design palettes available in palette.scss
// (imported above). For each palette, you can optionally specify a default, lighter, and darker
// hue.
$redlight-app-primary: mat-palette($mat-red);
$redlight-app-accent:  mat-palette($mat-pink, A200, A100, A400);

// The warn palette is optional (defaults to red).
$redlight-app-warn:    mat-palette($mat-red);

// Create the theme object (a Sass map containing all of the palettes).
$redlight-app-theme: mat-light-theme($redlight-app-primary, $redlight-app-accent, $redlight-app-warn);


//@include angular-material-theme($candy-app-theme);
$foreground: map-get($redlight-app-theme, foreground);
$background: map-get($redlight-app-theme, background);
$primary: map-get($redlight-app-theme, primary);
$accent: map-get($redlight-app-theme, accent);
$warn: map-get($redlight-app-theme, warn);


@mixin redlight-material-theme($theme) {
    // Extract whichever individual palettes you need from the theme.
    $primary: map-get($theme, primary);
    $accent: map-get($theme, accent);
    $background: map-get($theme, background);
    $foreground: map-get($theme, foreground);
}

This is what I have no. I assume I import this file into my componet.scss and then I can call the mixin?

**Comment by jaetea** [comment]
@jelbourn what does this comment mean?

The key takeaway is: don't import the file where you @include theming mixins. Anything else is fine to import.

Is this incorrect?

`@import "assets/css/themes/feelingblue";
@import "assets/css/themes/redlight";

@include angular-material-theme($candy-app-theme);

@include redlight-material-theme($redlight-app-theme);`
